tests/Integration/RequestIntegrationTest.php

```diff
@@ -8,52 +8,77 @@
+     * @dataProvider invokeUserContextPiiDataProvider
      */
-    public function testInvoke(array $requestData, array $expectedValue): void
+    public function testInvokeWithRequestHavingIpAddress(bool $pii, array $expectedValue): void
     {
         $event = new Event();
+        $event->getUserContext()->setData(['foo' => 'bar']);
 
         $request = new ServerRequest();
-        $request = $request->withCookieParams($requestData['cookies']);
-        $request = $request->withUri(new Uri($requestData['uri']));
-        $request = $request->withMethod($requestData['method']);
-
-        foreach ($requestData['headers'] as $name => $value) {
-            $request = $request->withHeader($name, $value);
-        }
+        $request = $request->withHeader('REMOTE_ADDR', '127.0.0.1');
 
         $this->assertInstanceOf(ServerRequestInterface::class, $request);
 
-        RequestIntegration::applyToEvent($event, $request);
+        $integration = new RequestIntegration(new Options(['send_default_pii' => $pii]));
+
+        RequestIntegration::applyToEvent($integration, $event, $request);
 
-        $this->assertEquals($expectedValue, $event->getRequest());
+        $this->assertEquals($expectedValue, $event->getUserContext()->toArray());
     }
 
-    public function testInvokeWithRequestHavingIpAddress(): void
+    public function invokeUserContextPiiDataProvider(): array
+    {
+        return [
+            [
+                true,
+                ['ip_address' => '127.0.0.1', 'foo' => 'bar'],
+            ],
+            [
+                false,
+                ['foo' => 'bar'],
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider invokeDataProvider
+     */
+    public function testInvoke(bool $pii, array $requestData, array $expectedResult): void
     {
         $event = new Event();
-        $event->getUserContext()->setData(['foo' => 'bar']);
 
         $request = new ServerRequest();
-        $request = $request->withHeader('REMOTE_ADDR', '127.0.0.1');
+        $request = $request->withCookieParams($requestData['cookies']);
+        $request = $request->withUri(new Uri($requestData['uri']));
+        $request = $request->withMethod($requestData['method']);
+
+        foreach ($requestData['headers'] as $name => $value) {
+            $request = $request->withHeader($name, $value);
+        }
+
         $this->assertInstanceOf(ServerRequestInterface::class, $request);
 
-        RequestIntegration::applyToEvent($event, $request);
+        $integration = new RequestIntegration(new Options(['send_default_pii' => $pii]));
+
+        RequestIntegration::applyToEvent($integration, $event, $request);
 
-        $this->assertEquals(['ip_address' => '127.0.0.1', 'foo' => 'bar'], $event->getUserContext()->toArray());
+        $this->assertEquals($expectedResult, $event->getRequest());
     }
 
     public function invokeDataProvider(): array
     {
         return [
             [
+                true, // send_default_pii
```

Comment:
In all data providers we never commented anything. If you think that the test case is not enough clear, a better way to describe it is to use as key a little phrase describing what it does. See the example below:

```php
return [
    'description' => [
        ...
    ]
];
```

You can also mix test cases that have a name with cases that don't have one, even though I don't think is nice to see 😄